### PR TITLE
Add optional text input component for list item

### PR DESCRIPTION
**Pull request by Joshuabaker2**
This addresses #260, giving the ability to have a text input component within the listitem component.

### CODE REVIEW

File: src/list/ListItem.js [hunk 1]
```diff
@@ -222,8 +278,11 @@ styles = StyleSheet.create({
     marginRight: 5,
     color: colors.grey4
   },
-  chevron: {
-  }
+  textInputStyle: {
+    height: 20,
+    textAlign: 'right'
+  },
+  chevron: {}
```
Comment (binoy14) [inline src/list/ListItem.js]: @Joshuabaker2 small thing but can you remove unused styles. Rest all looks good 👍

**Comment by Joshuabaker2** [comment]
Unused styles removed & commits squashed!

**Comment by binoy14** [comment]
@Joshuabaker2 Your awesome! Thanks for the PR

**Comment by alien3d** [comment]
I'm just checking here for the same thing..  How do i use it ?  I have manually changed the listitem in node module..  I write like this  "          <ListItem title="Username"  textInput={true}  textInputKeyboardType="default"></ListItem>" seem nothing happen

**Comment by binoy14** [comment]
@Joshuabaker2 any thoughts on this?

**Comment by Joshuabaker2** [comment]
Sure thing! It's just like the [React Native TextInput](https://facebook.github.io/react-native/docs/textinput.html), in that you need to set the value while utilizing a callback. In the React Native TextInput props, it is using an `onChangeText` callback and the `value` to set it (check out the docs) and for ours, since we are wrapping this component with a lot more stuff, the names were changed to `textInputOnChangeText` and `textInputValue` respectively. 

As a really basic example, it would look like: 

```
...
      <ListItem
        textInputOnChangeText={(text) => this.setState({text})}
        textInputValue ={this.state.text}
      />
...
```

**Comment by Monte9** [comment]
@Joshuabaker2 It would be helpful for everyone if you added this example to the readme.

**Comment by alien3d** [comment]
Agree .. I would love to see good example  basic login..  @Monte9  ..

**Comment by alien3d** [comment]
@Joshuabaker2 How do you solve focus issue? since refs cannot be access?
[test file](https://pastebin.com/xUjt1fd7)
List item change
[List item](https://pastebin.com/sxT8pBTXl)
@Monte9  still incomplete example :(

**Comment by alien3d** [comment]
Even thou close.. i'm still working on real example list view text input and now it's working. It may not the same with current file..
[test file](https://pastebin.com/dWZDVLdM)
[listitem](https://pastebin.com/d8UkLSVB)

A full sample login will be complete next weekend  :) .. login /register using react element with  php /c#

**Comment by Monte9** [comment]
@alien3d awesome! Keep us updated and kindly post your solution back here. Thanks

**Comment by marksbren** [comment]
I cannot get the TextInput to display on the list item. The following code does not work for me. I set the this.state.newMax value to "20" to start but there is nothing showing on the right and the keyboard does not activate when a tap there.

```
<ListItem
	key={1}
	title="Maximum Value"
	hideChevron={true}
	textInput={true}
	textInputMaxLength={3}
	textInputKeyboardType='numeric'
	textInputValue={this.state.newMax}
	textInputSecureTextEntry={false}
	textInputOnChangeText={(text) => this.setState({newMax:text})}
/>
```
iphone 6 ios 10 3 14e269 2017-04-28 08-04-29

**Comment by alien3d** [comment]
Dear  @marksbren,  textInputSecureTextEntry i think not in the current path yet 
[Current file](https://raw.githubusercontent.com/react-native-training/react-native-elements/master/src/list/ListItem.js)

What my latest sample even not complete yet still can be used..

You may check out this before i release full functional with back end ..
[test file](https://gist.github.com/alien3d/7a2ff42647958ecee8e77085a99e6c21)
[listitem.js](https://gist.github.com/alien3d/1d3100ca43e643a8e8ab83a636415370)

**Pull request merged**